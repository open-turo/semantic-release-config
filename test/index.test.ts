import template = require("lodash.template");

import config from "~/index";

describe("Branches", () => {
  const prereleasesToTest: string[] = [];
  beforeAll(() => {
    // This date is the time of writing this test (2022-12-15 01:47PM PST)
    jest.useFakeTimers().setSystemTime(1_671_140_834_629);
    for (const branchConfig of config.branches) {
      if (
        typeof branchConfig === "object" &&
        "prerelease" in branchConfig &&
        typeof branchConfig.prerelease === "string"
      ) {
        prereleasesToTest.push(branchConfig.prerelease);
      }
    }
  });

  afterAll(() => {
    jest.useRealTimers();
  });

  test.each([
    "f/test",
    "f/project-1234_test_that_semantic_release_works",
    "c/this_is_a_branch_that_does_not-follow-any-convention-223434",
  ])(
    "generates the right prerelease names for Turo custom branches: %s",
    (branch) => {
      for (const prerelease of prereleasesToTest) {
        expect(template(prerelease)({ name: branch })).toMatchSnapshot();
      }
    }
  );
});
