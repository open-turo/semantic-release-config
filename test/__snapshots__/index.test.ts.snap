// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Branches generates the right prerelease names for Turo custom branches: c/this_is_a_branch_that_does_not-follow-any-convention-223434 1`] = `"beta-c-this-is-a-branch-that-does-not-follow-any-convention-223434-1671140834629"`;

exports[`Branches generates the right prerelease names for Turo custom branches: f/project-1234_test_that_semantic_release_works 1`] = `"beta-project-1234-1671140834629"`;

exports[`Branches generates the right prerelease names for Turo custom branches: f/test 1`] = `"beta-f-test-1671140834629"`;
